x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/analytics/AnalyticsTemplateRegistry.java

```diff
@@ -119,4 +123,9 @@ protected Map<String, ComponentTemplate> getComponentTemplateConfigs() {
     protected Map<String, ComposableIndexTemplate> getComposableTemplateConfigs() {
         return COMPOSABLE_INDEX_TEMPLATES;
     }
+
+    @Override
+    protected boolean requiresMasterNode() {
```

Comment:
Can you add a comment to explain why it is needed?

Reply:
Added some comments.

Reply:
thanks